README.rst

```diff
@@ -167,12 +168,85 @@ Additionally Supported Parameters
 |                 |                                                                             |
 |                 | Defaults to the truncated description.                                      |
 +-----------------+-----------------------------------------------------------------------------+
-|       url       | Adds a URL in email as <br/>                                                | 
+|       url       | Adds a URL in email as <br/>                                                |
 |                 |                                                                             |
 |                 | ``For more information see <url>.`` <br/>                                   |
-|                 |                                                                             | 
+|                 |                                                                             |
 |                 | Default to **None** meaning above message would not be added to the email.  |
 +-----------------+-----------------------------------------------------------------------------+
+|       type      | Set values of other parameters based on predefined setting                  |
+|                 | ``OPENWISP_NOTIFICATION_TYPES``                                             |
+|                 |                                                                             |
+|                 | Defaults to **None** meaning you need to provide other arguments.           |
++-----------------+-----------------------------------------------------------------------------+
+
+Settings
+--------
+
+``OPENSWISP_NOTIFICATION_TYPES``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    
+
++--------------+-------------+
+| **type**:    | ``dict``    |
++--------------+-------------+
+| **default**: | ``{}``      |
++--------------+-------------+
+
+This setting allows to define additional notification types.
+
+For example, if you want to add a notification type ``device added`` you can use:
+
+.. code-block:: python
+
+    # In your_project/settings.py
+    OPENSWISP_NOTIFICATION_TYPES = {
+        'custom type': {
+            'level': 'info',
+            'verb': 'added',
+            'name': 'device added',
+            'message': 'configurables/message.md',
+        },
+    }
+
+``OPENWISP_NOTIFICATION_MESSAGE_TEMPLATE``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+--------------------------------+
+| **type**:    | ``string``                     |
++--------------+--------------------------------+
+| **default**: | ``configurables/message.md``   |
++--------------+--------------------------------+
+
+This setting allows to provide a default markdown formatted message template.
```

Comment:
I think there was a misunderstanding here.

We need 2 things:

1. a general email template, for styling emails
2. each notification type should have its own message (which should be displayed both in email and web notifications) and a subject (the subject should be used in the email subject and can be used the list of the notifications in the JS widget), which can be markdown strings, I think these can go in the `NOTIFICATION_TYPES`

Reply:
The words I used might be not proper to convey this. I wanted to say that this setting will provide a template for notification message, not for email. I will try to emphasize that.